docs/build/building-apps/06-app-go-genesis.md

```diff
@@ -0,0 +1,54 @@
+---
+sidebar_position: 1
+---
+
+
+### Modifying the `DefaultGenesis`
+
+It is possible to modify the DefaultGenesis parameters for modules by wrapping the module, providing it to the `*module.Manager` and injecting it with `depinject`.
+
+
+Example ( staking ) :
+
+```go
+type CustomStakingModule struct {
+    staking.AppModule
+    cdc codec.Codec
+}
+
+// DefaultGenesis will override the Staking module DefaultGenesis AppModuleBasic method.
+func (cm CustomStakingModule) DefaultGenesis() json.RawMessage {
+    params := stakingtypes.DefaultParams()
+    params.BondDenom = "mydenom"
+
+    return cm.cdc.MustMarshalJSON(&stakingtypes.GenesisState{
+        Params: params,
+    })
+}
+
+// option 1: use new module manager
+moduleManager := module.NewManagerFromMap(map[string]appmodule.AppModule{
+    stakingtypes.ModuleName: CustomStakingModule{cdc: appCodec, AppModule: staking.NewAppModule(...)},
+	// other modules ...
+})
+
+// option 2: override previous module manager
+oldStakingModule,_ := moduleManager.Modules()[stakingtypes.ModuleName].(staking.AppModule)
+moduleManager.Modules()[stakingtypes.ModuleName] = CustomStakingModule{
+	AppModule: oldStakingModule,
+	cdc: appCodec,
+}
+
+
+// depinject users
```

Comment:
I think the flow is backwards:
- first you inject a module manager (or have it already without depinject)
- then you modify it (option 2 for depinject users, option 1 for non depinject users)
- then you set it on the app (missing here)

Reply:
Since moduleManager is a pointer and the map is a reference there's no need to override the whole MM. wdyt ?